app/models/firm.rb

```diff
@@ -1,3 +1,96 @@
 class Firm < ActiveRecord::Base
+  has_and_belongs_to_many :in_person_advice_methods
+  has_and_belongs_to_many :other_advice_methods
+  has_and_belongs_to_many :initial_advice_fee_structures
+  has_and_belongs_to_many :ongoing_advice_fee_structures
+  has_and_belongs_to_many :allowed_payment_methods
+  has_and_belongs_to_many :investment_sizes
+
+  belongs_to :initial_meeting_duration
+
   has_many :advisers
+
+  attr_accessor :percent_total
+
+  validates :email_address,
+    presence: true,
+    length: { maximum: 50 },
+    format: { with: /.+@.+\..+/ }
+
+  validates :telephone_number,
+    presence: true,
+    length: { maximum: 30 },
+    format: { with: /\A[0-9 ]+\z/ }
+
+  validates :address_line_one,
+    presence: true,
+    length: { maximum: 100 }
+
+  validates :address_line_two,
+    length: { maximum: 100 }
+
+  validates :address_postcode,
+    presence: true,
+    format: { with: /\A[a-zA-Z\d]{1,4} [a-zA-Z\d]{1,3}\z/ }
+
+  validates :address_town,
+    :address_county,
+    presence: true
+
+  validates :in_person_advice_methods,
+    length: { minimum: 1 }
+
+  validates :free_initial_meeting,
+    inclusion: { in: [true, false] }
+
+  validates :initial_meeting_duration,
+    presence: true,
+    if: ->{ free_initial_meeting }
```

Comment:
You didn't address my comment here before `free_initial_meeting?` works here and is better.